library/std/src/sys/windows/c.rs

```diff
@@ -511,6 +513,8 @@ pub struct REPARSE_DATA_BUFFER {
     pub rest: (),
 }
 
+/// NB: Use carefully! In general using this as a reference is likely to get the
+/// provenance wrong for the `PathBuffer` field!
```

Comment:
This is fine but it is a shame we need to keep repeating variations of this comment. But eh, it's fine. Encapsulating the unsafely is what the higher level APIs are for.

Reply:
Beats someone missing it -- by repeating it, it shows up in auto-complete in rust-analyzer.